hl7/4.2/modules/ROOT/edits/index.adoc

```diff
@@ -0,0 +1,174 @@
+This structure is a hierarchy of Java Maps and Lists. You use either DataWeave or code to manipulate them.
+Each transaction structure is defined in the schemas.
+
+The message itself contains the following keys, some of which apply only to either the read operation or the write operation, as indicated:
+
+[%header%autowidth.spread]
+|===
+|Key name |Description
+|ACK (read only) |Message generated by the module during the read operation. The `MSA-01` acknowledgment code value is based on the parser configuration settings.
+|Data (read or write) |Wrapper for message data, with a key matching the message structure ID value that links to the actual data. Allows different messages to be included in the metadata and handled in DataWeave mappings.
+|Delimiters (read or write) |The delimiters used for the message.
+The characters in the string of delimiters are interpreted based on their position in string, in the following order: component separator, repetition separator, escape character, and subcomponent separator, with a value of `U` indicating no value in that position.
+|Errors (read only) |A list of errors associated with the input message.
+|Id |Message structure ID
+|MSH (read only) |Link that receives MSH segment data
+|Name (read only) |Message structure name
+|===
+
+Individual messages have their own maps, with keys matching the segments of the message. For example, an ACK message uses the message structure ID ACK, and the data for the sent or received ACK message is an ACK value in the Data map. The ACK message is itself a map, and the segments and groups of the message are represented as maps (in the case of singleton instances) or lists of maps (for repeating instances) with positional keys.
+
+There are two special cases where using generic handling for data that is not included in a schema definition.
+
+The first is for HL7 values of the `varies` type. Since these values consist of any structure of components and subcomponents that can be repeated, the parser uses a list of map representations for each `varies` of type. The keys in each map are generated as the value is parsed, matching standard HL7 value names with two digits used for each nesting level.
+So a simple text value for an `OBX-05 Observation Value` field, would use the key `OBX-05` in a map. If there are two components present, they use keys `OBX-05-01` and `OBX-05-02`.
+
+The second is extension segments with tags that match a pattern configured under parser options. These are similar in structure to the `varies` values, with the exception of only being within a single map for the entire segment.
+
+The maps containing extension segment data are added to the basic message map in lists with the key `ExtensionSegs`. In addition to the actual extension segment data, the map for the extension segment contains two other keys:
+
+[%header%autowidth.spread]
+|===
+|Key |Description
+|Ident |The extension segment identifier (tag)
```

Comment:
Must end with a period because its peer does.
Again, try to make peers parallel in sentence structure.